File: php/laravel/src/Enums/TraceStatus.php
<?php

namespace Smartness\TraceFlow\Enums;

enum TraceStatus: string
{
    case PENDING = 'PENDING';
    case RUNNING = 'RUNNING';
    case SUCCESS = 'SUCCESS';
    case FAILED = 'FAILED';
    case CANCELLED = 'CANCELLED';
}

File: java/core/src/main/java/com/smartness/traceflow/enums/TraceStatus.java
package com.smartness.traceflow.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum TraceStatus {
    PENDING("PENDING"),
    RUNNING("RUNNING"),
    SUCCESS("SUCCESS"),
    FAILED("FAILED"),
    CANCELLED("CANCELLED");

    private final String value;

    TraceStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}
